### ISSUE TITLE
Option to have the query and mutation classes extend another interface

### ISSUE THREAD

**Issue by benmccann** [kobylynskyi/graphql-java-codegen#135]
The example server has `public class QueriesResolver implements Query, GraphQLQueryResolver`. What if `Query` extended `GraphQLQueryResolver`? Then you wouldn't have to implement two interfaces on each resolver. We could have a new option that let you specify a fully qualified class name to have the queries extend. Same for the mutations.

**Comment by joffrey-bion** [comment]
Hi Ben!
I also wanted to suggest this at first, but the problem is that `GraphQLQueryResolver` is an interface from [GraphQL Java Tools](https://github.com/graphql-java-kickstart/graphql-java-tools), it's declared [here](https://github.com/graphql-java-kickstart/graphql-java-tools/blob/14afd845af8873b7dd0c140b0919dd050de7ba0b/src/main/kotlin/graphql/kickstart/tools/GraphQLQueryResolver.kt).

It is not desirable for this library to depend on GraphQL Java Tools, because it introduces an unnecessary dependency for people who are not using that lib.

This codegen library does generate code that happens to follow the conventions of GraphQL Java Tools, so that it's easy to use the 2 libs together. However, it currently does not force users to depend on it.

**Comment by kobylynskyi** [comment]
@benmccann, thanks for creating the feature request. I was also thinking about this some time ago.

I very like the solution from @joffrey-bion with one minor comment. I would make the nested config so that in Maven it will look a bit prettier:

```xml
<parentInterfaces>
    <queryResolver>graphql.kickstart.tools.GraphQLQueryResolver</queryResolver>
    <mutationResolver>graphql.kickstart.tools.GraphQLMutationResolver</mutationResolver>
    <subscriptionResolver>graphql.kickstart.tools.GraphQLSubscriptionResolver</subscriptionResolver>
    <resolver>graphql.kickstart.tools.GraphQLResolver<{{TYPE_NAME}}></resolver>
</parentInterfaces>
```

**Comment by benmccann** [comment]
I like that solution and the placeholder. I wonder if it should be `${TYPE_NAME}` for the placeholder because that might be able to be used directly in FreeMarker and will be familiar to folks who have used bash variables

**Comment by kobylynskyi** [comment]
We can go with `${TYPE_NAME}` too, but I don't think we will use it in the FreeMarker template directly because typeName can have prefix/suffix as well.
Imagine the following configuration:

```xml
<modelNameSuffix>TO</modelNameSuffix>
<parentInterfaces>
    <resolver>graphql.kickstart.tools.GraphQLResolver<${TYPE_NAME}></resolver>
</parentInterfaces>
```

So the resulting generated Resolver interface will look like:

```java
public interface AcceptTopicSuggestionPayloadResolver implements graphql.kickstart.tools.GraphQLResolver<GithubAcceptTopicSuggestionPayloadTO> {

    GithubTopicTO topic(GithubAcceptTopicSuggestionPayloadTO githubAcceptTopicSuggestionPayloadTO, graphql.schema.DataFetchingEnvironment env) throws Exception;

}
```

**Comment by joffrey-bion** [comment]
I have actually avoided the `$` syntax on purpose, because I'm afraid it will collide with the Maven/Gradle plugin configuration syntax.
Both of them support some sort of variables in the build files, and AFAIR `$` has a meaning in both.

**Comment by benmccann** [comment]
Ah, good call. Yeah, probably better to avoid `${TYPE}` given the complication it might create with the build system.

**Comment by kobylynskyi** [comment]
Released in [1.8.0](https://github.com/kobylynskyi/graphql-java-codegen/releases/tag/v1.8.0)
